### PR TITLE
Create RedPajama-3B.md

### PR DESCRIPTION
**Pull request by adamsch1**

### CODE REVIEW

File: docs/RedPajama-3B.md
Comment (csris) [inline docs/RedPajama-3B.md:1]: Let's rename this file to `finetuning-RedPajama-3B.md`

File: docs/RedPajama-3B.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# RedPajama-3B
```
Comment (csris) [inline docs/RedPajama-3B.md:1]:
```suggestion
# Fine Tuning RedPajama-INCITE-Base-3B
```

File: docs/RedPajama-3B.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+In order to fine-tune the RedPajama 3B models, please follow these steps:
```
Comment (csris) [inline docs/RedPajama-3B.md:3]:
```suggestion
In order to fine-tune the Fine Tuning RedPajama-INCITE-Base-3B model, please follow these steps:
```

File: docs/RedPajama-3B.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+This script will download the weight from HuggingFace and prepare it for finetuning. The prepared weights will be saved at 
+
+```
+pretrained/RedPajama-3B/togethercomputer_RedPajama-INCITE-Chat-3B-v1
+```
+
+# Prepare Fine Tuning Data
+
+We now need to preapre the training data.  We provide an example script that downloads a small slice of data from OIG.
```
Comment (csris) [inline docs/RedPajama-3B.md:27]:
```suggestion
We now need to prepare the training data.  We provide an example script that downloads a small slice of data from OIG. 
```

File: docs/RedPajama-3B.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# RedPajama-3B
+
```
Comment (csris) [inline docs/RedPajama-3B.md:2]: Let's add an introductory paragraph here.

> In this tutorial, you will learn how to fine-tune a base LLM on a sample of data. By the end of the tutorial, you will have fine-tuned the RedPajama-INCITE-Base-3B model using a sample of chat data from the OIG dataset. You can adapt this tutorial to fine-tune with your own data.

File: docs/RedPajama-3B.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+To download this sample dataset, please run:
+ 
+```
+bash data/OIG-chip2/prepare.sh
+````
+ 
+The sample dataset will be saved at 
+
+```
+data/OIG-chip2/unified_chip2.jsonl.
```
Comment (csris) [inline docs/RedPajama-3B.md:37]:
```suggestion
data/OIG-chip2/unified_chip2.jsonl
```

File: docs/RedPajama-3B.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+```
+
+# Convert to Huggingface Format
+
+Convert to HF format. The fine-tuned model will be saved to 
+
+```
+model_ckpts/rp-incite-chat-3b-finetuned/checkpoint_{steps}
+```
+
```
Comment (csris) [inline docs/RedPajama-3B.md:56]: Please add a sentence that says what the the default number of steps is.

File: docs/RedPajama-3B.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+In order to use it for inference, you will need to convert it to the HuggingFace format. To do so, run the following script 
+(as an example, please change the checkpoint path, n-stages and n-layer-per-stage according to the training script):
+
+```
+python tools/convert_to_hf_gptneox.py --config-name togethercomputer/RedPajama-INCITE-Chat-3B-v1 --ckpt-path model_ckpts/rp-incite-chat-3b-fintuned/checkpoint_100/ --save-path model_ckpts/hf --n-stages 4 --n-layer-per-stage 8
```
Comment (csris) [inline docs/RedPajama-3B.md:61]:
```suggestion
python tools/convert_to_hf_gptneox.py 
  --config-name togethercomputer/RedPajama-INCITE-Chat-3B-v1 
  --ckpt-path model_ckpts/rp-incite-chat-3b-fintuned/checkpoint_100/ 
  --save-path model_ckpts/hf 
  --n-stages 4 
  --n-layer-per-stage 8
```

**Review by justusc — APPROVED** [review]

**Review by csris — APPROVED** [review]

**Pull request merged**